inception-ui-kb/src/main/java/de/tudarmstadt/ukp/inception/ui/kb/value/editor/IRIValuePresenter.java

```
@@ -17,44 +17,95 @@
+        LambdaAjaxLink link = new LambdaAjaxLink("link",
+            t -> actionIRILinkClicked(t, getKBObject()));
+        link.add(new Label("label", Model.of(getLabel(getKBObject()))));
         add(link);
     }
 
-    private String getLabel()
+    Optional<KBObject> getKBObject() {
+        Object stmtValue = getModelObject().getValue();
+        Optional<KBObject> kbObject = kbService
+            .readKBIdentifier(kbModel.getObject(), stmtValue.toString());
+        if (kbObject.isPresent()) {
+            return kbObject;
+        }
+        return Optional.empty();
+    }
+
+    private String getLabel(Optional<KBObject> aKbObject)
     {
-        KBStatement stmt = getModelObject();
-        return ((IRI) stmt.getValue()).getLocalName();
+        if (aKbObject.isPresent()) {
+            return aKbObject.get().getUiLabel();
+        }
+        return ((IRI) getModelObject().getValue()).getLocalName();
     }
 
-    private void actionIRILinkClicked(AjaxRequestTarget target)
+    private void actionIRILinkClicked(AjaxRequestTarget aTarget, Optional<KBObject> aKbObject)
     {
-        // TODO need to know what the IRI refers to - concept, property, both???
-        KBStatement stmt = getModelObject();
-        KBHandle selectedConcept = new KBHandle(((IRI) stmt.getValue()).toString());
-        send(getPage(), Broadcast.BREADTH,
-                new AjaxConceptSelectionEvent(target, selectedConcept, true));
+        if (aKbObject.isPresent()) {
+            KBObject kbObject = aKbObject.get();
+            if (kbObject instanceof KBConcept) {
+                send(getPage(), Broadcast.BREADTH,
+                    new AjaxConceptSelectionEvent(aTarget, KBHandle.of(kbObject), true));
+            }
+            else if (kbObject instanceof KBInstance) {
+                send(getPage(), Broadcast.BREADTH,
+                    new AjaxInstanceSelectionEvent(aTarget, KBHandle.of(kbObject)));
+            }
+            else {
```

Could we do a `instanceof KBProperty` here and throw an exception if the KBObject is an unknown type of instance?